### DISCUSSION TITLE
Filtering + Expanding doesn't work for subRows

### DISCUSSION THREAD

**Discussion by Steffi3rd** [TanStack/table#4899] [Q&A]
Hello 👋

I would like to how to apply filter for `subRows`.

On this official "Expanding" example  of TanStack : https://jr4nn3.sse.codesandbox.io
When we use filter, this filter only work for first the level (depth: 0)

If the depth: 1 or depth: 2 include my filter value, it will not appear.

How to fix this please?

**Comment by Steffi3rd** [comment]
Solution is here : `filterFromLeafRows: true ` : https://github.com/TanStack/table/discussions/3828#discussioncomment-2586676

```
  const table = useReactTable({
    data,
    columns,
    // ...
    filterFromLeafRows: true,
  });
```

Docs : https://tanstack.com/table/v8/docs/api/features/filters#filterfromleafrows
**By default, filtering is done from parent rows down (so if a parent row is filtered out, all of its children will be filtered out as well).** Setting this option to true will cause filtering to be done from leaf rows up (which means parent rows will be included so long as one of their child or grand-child rows is also included).

**Reply by alexlazarian** [reply]
According to the documentation [here](https://tanstack.com/table/v8/docs/api/features/filters#filterfromleafrows), enabling the filterFromLeafRows option will include parent rows if any of their child or grand-child rows are included in the filter results.

However, when I enable this option, I’ve noticed that when searching for the parent row itself, the children rows are filtered out, which isn’t the desired behavior. I would like to preserve the visibility of subrows even when searching for the parent row.

Is there a way to adjust the filtering behavior so that subrows are preserved and visible when searching for the parent row?

Any insights or suggestions on how to achieve this would be greatly appreciated!
@tannerlinsley
